lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +48,46 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        meta_files = list(self.path_to_raw_txt_data.glob('*meta.json'))
+        raw_files = list(self.path_to_raw_txt_data.glob('*raw.txt'))
+        if len(meta_files) != len(raw_files):
+            raise InconsistentDatasetError
+
+        for path in (meta_files + raw_files):
+            if int(path.stem.split('_')[0]) > len(meta_files):
+                raise InconsistentDatasetError
+
+        for file in (meta_files + raw_files):
```

Comment:
join two loops, they have same iterable, throw same exception, what is different is the check